src/test/java/StringManipulationTest.java

```diff
@@ -50,100 +69,194 @@ public void testRemoveNthCharacter1() {
     @Test
     public void testRemoveNthCharacter2() {
         manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
-        assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?", manipulatedstring.removeNthCharacter(3, true));
+        assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?",
+                manipulatedstring.removeNthCharacter(3, true));
     }
 
+    // Tests string with newline characters
     @Test
     public void testRemoveNthCharacter3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Seeing what happens when\nnewline characters are introduced"); // Newline character counts as character
+        assertEquals("See ng  hat hap ens whe \nne lin  ch rac ers are int odu ed", manipulatedstring.removeNthCharacter(4, true));
     }
 
+    // Tests large value for n
     @Test
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("What happens when n is a much larger value?");
+        assertEquals("What happns when nis a muchlarger vaue?", manipulatedstring.removeNthCharacter(10, false));
     }
 
+    // Tests n = 1 (all characters should be replaced with a space)
     @Test
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("What happens if n is 1?");
+        assertEquals("                       ", manipulatedstring.removeNthCharacter(1, true));
     }
 
+    // Tests whether method throws error when n = 0
     @Test
     public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("An exception should be thrown if n = 0");
+        try {
+            manipulatedstring.removeNthCharacter(0, false);
+            fail("No exception thrown for n=0");
+        } catch (IllegalArgumentException e) {
+            assert true;
+        }
     }
 
+    // Tests whether method throws error when n > length
     @Test
```

Comment:
I shared a link below that talks about using Lamba expressions when dealing with exceptions in JUnit.